File: lecture010/incdec.cpp
#include <iostream>

using namespace std;
void inc_dec(int a, int b)
{
    if (a == b)
    {                      //OR
        cout << b << endl; //cout<<a;
        return;
    }

    cout << b;         //cout<<a;
    inc_dec(a, b - 1); //inc_dec(a+1,b)
    cout << b;         //cout<<a;
}

void main(int args, char **argv)
{
    inc_dec(1, 5);
}

File: lecture010/incdec.java
import java.util.Scanner;

public class incdec {
    public static Scanner scn = new Scanner(System.in);

    public static void main(String[] args) {
        solve(1, 6);
    }

    public static void solve(int a, int b) {
        if (a == b) {
            System.out.println(a);
            return;
        }

        System.out.println(a);
        // if(a==5) return ;
        solve(a + 1, b);

        System.out.println(a);

    }
}